lib/inferno/dsl/fhir_validation.rb

```diff
@@ -116,7 +116,16 @@ def exclude_message(&block)
         def resource_is_valid?(resource, profile_url, runnable)
           profile_url ||= FHIR::Definitions.resource_definition(resource.resourceType).url
 
-          outcome = FHIR::OperationOutcome.new(JSON.parse(validate(resource, profile_url)))
+          begin
+            validation_response = validate(resource, profile_url)
+          rescue StandardError => e
+            runnable.add_message('error', e.message)
+            raise Inferno::Exceptions::ErrorInValidatorException, "Unable to connect to validator at #{url}."
+          end
+
+          # The validator should return OperationOutcome both on successful validation and on error,
+          # so parse the OO before checking the HTTP status code.
+          outcome = FHIR::OperationOutcome.new(JSON.parse(validation_response.body))
```

Comment:
We should specifically handle the case of non-json responses and display those in the message without the json parse error.

Reply:
Updated to check for this specifically: ![image](https://github.com/inferno-framework/inferno-core/assets/13512036/f1632051-b20e-4c38-b533-72f6febc15f6)

Unfortunately this all put the `resource_is_valid?` function over rubocop's cyclomatic complexity limit and I had to refactor a bit. Feel free to say the refactorings aren't clear and I can spend a little more time on it